frontend/src/scenes/authentication/passwordResetLogic.ts

```diff
@@ -46,57 +41,77 @@ export const passwordResetLogic = kea<passwordResetLogicType>({
                 },
             },
         ],
-        newPasswordResponse: [
-            null as ResponseType | null,
+    })),
+    reducers({
+        requestPasswordResetSucceeded: [
+            false,
             {
-                updatePassword: async ({
-                    password,
-                    passwordConfirm,
-                }: {
-                    password: string
-                    passwordConfirm: string
-                }) => {
-                    if (!values.validatedResetToken?.token || !values.validatedResetToken.uuid) {
-                        return {
-                            success: false,
-                            errorCode: 'invalid_token',
-                            errorDetail: 'Your link is invalid or expired.',
-                        }
-                    }
-                    if (passwordConfirm !== password) {
-                        return {
-                            success: false,
-                            errorCode: 'confirmation_does_not_match',
-                            errorDetail: 'Password confirmation does not match.',
-                        }
-                    }
-                    try {
-                        await api.create(`api/reset/${values.validatedResetToken.uuid}/`, {
-                            password,
-                            token: values.validatedResetToken.token,
-                        })
-                        return { success: true }
-                    } catch (e: any) {
-                        return { success: false, errorCode: e.code, errorDetail: e.detail }
-                    }
-                },
+                submitRequestPasswordResetSuccess: () => true,
+            },
+        ],
+        passwordResetSucceeded: [
+            false,
+            {
+                submitPasswordResetSuccess: () => true,
             },
         ],
     }),
-    listeners: {
-        updatePasswordSuccess: async ({ newPasswordResponse }, breakpoint) => {
-            if (newPasswordResponse.success) {
-                lemonToast.success('Your password has been changed. Redirecting…')
-                await breakpoint(3000)
-                window.location.href = '/' // We need the refresh
-            }
+    forms(({ values, actions }) => ({
+        requestPasswordReset: {
+            defaults: {} as unknown as { email: string },
+            errors: ({ email }) => ({
+                email: !email ? 'Please enter your email to continue' : undefined,
+            }),
+            submit: async ({ email }, breakpoint) => {
+                await breakpoint()
```

Comment:
On its own without a number, as the first statement in a function, this shouldn't do anything... 🤔

Reply:
Is there something you would recommend for this sort of thing? I have to admit the "breakpoint" stuff has never truly landed nicely in my mental model.

Reply:
A breakpoint is just a function that throws in case the parent function is called again.

```ts
async function submit(_, breakpoint) {
  await delay(100)
  breakpoint()
  console.log('print bla')
}

submit()
submit()
```

This prints the log message just once. That's because while we were delaying, we called `submit` again. Now one breakpoint throws and stops the function's execution. The other one (from the later invocation) does not.

Reply:
Because delaying when checking for breakpoints is such a common thing, the code above can also be written as

```ts
async function submit(_, breakpoint) {
  await breakpoint(100)
  console.log('print bla')
}
```